### PR TITLE
Added key based locking factory

### PR DESCRIPTION
**Pull request by avasquez614**

### CODE REVIEW

File: utilities/src/main/java/org/craftercms/commons/concurrent/locks/AbstractWeakKeyBasedLockFactory.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package org.craftercms.commons.concurrent.locks;
```
Comment (sumerjabri) [inline utilities/src/main/java/org/craftercms/commons/concurrent/locks/AbstractWeakKeyBasedLockFactory.java]: Copyright?

File: utilities/src/main/java/org/craftercms/commons/concurrent/locks/KeyBasedLockFactory.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.craftercms.commons.concurrent.locks;
```
Comment (sumerjabri) [inline utilities/src/main/java/org/craftercms/commons/concurrent/locks/KeyBasedLockFactory.java]: Copyright?

File: utilities/src/main/java/org/craftercms/commons/concurrent/locks/WeakKeyBasedReentrantLockFactory.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package org.craftercms.commons.concurrent.locks;
```
Comment (sumerjabri) [inline utilities/src/main/java/org/craftercms/commons/concurrent/locks/WeakKeyBasedReentrantLockFactory.java]: Copyright?

File: utilities/src/main/java/org/craftercms/commons/concurrent/locks/AbstractWeakKeyBasedLockFactory.java (hunk 1 above)
Comment (avasquez614) [reply utilities/src/main/java/org/craftercms/commons/concurrent/locks/AbstractWeakKeyBasedLockFactory.java]: Fixed

File: utilities/src/main/java/org/craftercms/commons/concurrent/locks/KeyBasedLockFactory.java (hunk 2 above)
Comment (avasquez614) [reply utilities/src/main/java/org/craftercms/commons/concurrent/locks/KeyBasedLockFactory.java]: Fixed

File: utilities/src/main/java/org/craftercms/commons/concurrent/locks/WeakKeyBasedReentrantLockFactory.java (hunk 3 above)
Comment (avasquez614) [reply utilities/src/main/java/org/craftercms/commons/concurrent/locks/WeakKeyBasedReentrantLockFactory.java]: Fixed

**Pull request merged**